### PR TITLE
progressive growing of optimization models

### PR DESCRIPTION
**Pull request by teytaud**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the **CONTRIBUTING** document and completed the CLA (see **CONTRIBUTING**).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/benchmark/experiments.py [hunk 1]
```diff
@@ -563,7 +563,9 @@ def fastgames(seed: tp.Optional[int] = None) -> tp.Iterator[Experiment]:
     seedg = create_seed_generator(seed)
     algos = ["NaiveTBPSA", "ScrHammersleySearch", "PSO", "OnePlusOne",
              "CMA", "TwoPointsDE", "QrDE", "LhsDE", "Zero", "StupidRandom", "RandomSearch", "HaltonSearch",
-             "RandomScaleRandomSearch", "MiniDE", "SplitOptimizer5", "NGO", "Shiva", "DiagonalCMA"]
+             "RandomScaleRandomSearch", "MiniDE", "SplitOptimizer5", "NGO", "Shiva", "DiagonalCMA",
```
Comment (jrapin) [inline nevergrad/benchmark/experiments.py]: Eventually, if they are used for only one experiment, there is no need creating them in experimental variants, you can directly create them for the experiment:

```python
algos += [ConfSplitOptimizer(num_optims=num, progressive=True).set_name(f"ProgOptimizer{num}") for num in [3, 5, 9, 13]]
```

File: nevergrad/optimization/optimizerlib.py [hunk 2]
```diff
@@ -797,6 +808,7 @@ class ConfSplitOptimizer(base.ConfiguredOptimizer):
         number of optimizers
     num_vars: optional list of int
         number of variable per optimizer.
+    progressive: optional bool
```
Comment (jrapin) [inline nevergrad/optimization/optimizerlib.py]: please add a definition for this parameter

File: nevergrad/optimization/optimizerlib.py [hunk 3]
```diff
@@ -806,8 +818,11 @@ def __init__(
         num_optims: int = 2,
         num_vars: tp.Optional[tp.List[int]] = None,
         multivariate_optimizer: base.ConfiguredOptimizer = CMA,
-        monovariate_optimizer: base.ConfiguredOptimizer = RandomSearch
+        monovariate_optimizer: base.ConfiguredOptimizer = RandomSearch,
+        progressive: bool = False
     ) -> None:
+        if progressive:  # The progressive setting is typically for noisy optimization, hence we switch to a noisy optimizer.
+            multivariate_optimizer = OptimisticNoisyOnePlusOne
```
Comment (jrapin) [inline nevergrad/optimization/optimizerlib.py]: I'm a bit afraid of this kind of change without it being explicit in the docstring

File: nevergrad/optimization/optimizerlib.py [hunk 4]
```diff
@@ -753,6 +757,8 @@ def __init__(
 
             assert self.num_vars[i] >= 1, "At least one variable per optimizer."
             self.parametrizations += [p.Array(shape=(self.num_vars[i],))]
+            for param in self.parametrizations:
```
Comment (jrapin) [inline nevergrad/optimization/optimizerlib.py]: This is were the problem was, the optimizers did not share the same random state

File: nevergrad/benchmark/experiments.py (hunk 1 above)
Comment (teytaud) [reply nevergrad/benchmark/experiments.py]: Done, thx.

File: nevergrad/optimization/optimizerlib.py (hunk 4 above)
Comment (teytaud) [reply nevergrad/optimization/optimizerlib.py]: Thx!

File: nevergrad/optimization/optimizerlib.py (hunk 2 above)
Comment (teytaud) [reply nevergrad/optimization/optimizerlib.py]: Done, thx.

File: nevergrad/optimization/optimizerlib.py (hunk 3 above)
Comment (teytaud) [reply nevergrad/optimization/optimizerlib.py]: Fixed, thx.

File: nevergrad/benchmark/experiments.py (hunk 1 above)
Comment (teytaud) [reply nevergrad/benchmark/experiments.py]: Actually the way the code is structured means that we need a str, not an algorithm.

Comment (jrapin) [reply nevergrad/benchmark/experiments.py]: What do you mean? The code accept both, the typing may complain but nevermind this.
Also, when I say "eventually" it meant not necessarily now, we'll discuss this later ;)

**Pull request merged**